0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1201 @@
+
+#### Response Format
+
+A successful result uses the HTTP response code **200 OK**. The body contains a JSON object with the following field:
+
+| Field | Value | Description |
+|---|---|---|
+| `fulfillment` | String | The Transfer's [Crypto-Condition Fulfillment][] in text format. |
+
+#### Example
+
+Request:
+
+```
+GET /transfers/3a2a1d9e-8640-4d2d-b06c-84f2cd613204/fulfillment
+```
+
+Response:
+
+```json
+HTTP/1.1 200 OK
+
+{
+  "fulfillment": "cf:0:_v8"
+}
+```
+
+#### Errors
+
+- [NotFoundError][]
+- [InvalidUriParameterError][]
+
+
+### Get Account
+[Get Account]: #get-account
+
+Get an account resource.
+
+**Authorization:** The owner of an account MUST be able to get the account resource. A ledger MAY allow other account owners or unauthorized users to get other account resources. A ledger MAY return a subset of fields when it returns an account resource to any client other than the account owner (for example, omitting the balance).
+
+#### Request Format
+
+GET the path from the `debit_account` or `credit_account` field of a transfer.
+
+**Metadata URL Name:** `account`. Suggested generic path: `/accounts/{name}`
+
+#### Response Format
+
+A successful response uses the HTTP response code **200 OK** and contains the [Account resource][] in JSON format.
+
+#### Example
+
+Request:
+
+```
+GET /accounts/bob
+```
+
+Response:
+
+```json
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "id": "https://red.ilpdemo.org/ledger/accounts/bob",
+  "name": "bob",
+  "balance": "5007734.0",
+  "ledger": "https://red.ilpdemo.org/ledger",
+  "fingerprint": "88:90:3a:e7:e5:1c:c4:51:05:4b:0c:2b:3f:41:df:bf:0c:21:f3:78",
+  "minimum_allowed_balance": "-infinity"
+}
+```
+
+#### Errors
+
+- [NotFoundError][]
+- [InvalidUriParameterError][]
+
+
+### Send Message
+[Send Message]: #send-message
+
+Try to send a notification to another account. The message is only delivered if the other account is subscribed to [account notifications](#subscribe-to-account) on a WebSocket connection. ILP Clients and ILP Connectors use this method to request and send quotes.
+
+**Authorization:** The `from` field of the message MUST match the account owner. Unauthenticated users MUST NOT be able to send messages. A ledger MAY allow admin connections to send messages whose `from` value matches the ledger's base [URL][].
+
+#### Request Format
+
+POST a JSON [Message resource][] to the `message` [URL][].
+
+**Metadata URL Name:** `message`. Suggested generic path: `message`
+
+#### Response Format
+
+A successful response uses the HTTP response code **204 No Content** and contains no message body.
+
+#### Example
+
+Request:
+
+```json
+POST /messages
+Content-Type: application/json
+
+{
+  "from": "https://blue.ilpdemo.ripple.com/ledger/accounts/bob",
+  "to": "https://blue.ilpdemo.ripple.com/ledger/accounts/alice",
+  "ledger": "https://blue.ilpdemo.ripple.com/ledger",
+  "data": {
```

Comment:
This question is tricky. The size of the message data depends on the encoding. Any limit I could think of, upper or lower, would be totally arbitrary. Maybe we should decide on a minimum limit in bytes and/or characters of UTF-8, chosen based on being able to convey any reasonable quote / ILP header data we expect to need?